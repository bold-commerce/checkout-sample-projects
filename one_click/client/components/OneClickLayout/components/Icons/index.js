export { default as TimesCircle } from './TimesCirlce'
export { default as ArrowRight } from './ArrowRight'
export { default as CaretRight } from './CaretRight'