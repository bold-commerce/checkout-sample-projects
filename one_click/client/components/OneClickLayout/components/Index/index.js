export { default as IndexPage } from "./IndexPage";
export { default as IndexGuest } from "./IndexGuest";