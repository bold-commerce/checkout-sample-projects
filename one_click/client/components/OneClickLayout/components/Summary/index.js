export { default as Summary } from './Summary';
export { default as SummaryCondensed } from './SummaryCondensed';