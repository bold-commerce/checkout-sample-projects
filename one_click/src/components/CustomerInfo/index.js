export { default as CustomerInfo } from './CustomerInfo'