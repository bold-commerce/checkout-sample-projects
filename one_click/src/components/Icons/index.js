export { default as ChevronRight } from './ChevronRight'
export { default as TimesCircle } from './TimesCircle'
export { default as ArrowRight } from './ArrowRight'
export { default as CaretRight } from './CaretRight'
export { default as Times } from './Times'
export { default as Plus } from './Plus'