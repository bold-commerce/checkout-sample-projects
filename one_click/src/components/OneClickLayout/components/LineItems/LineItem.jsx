import React from 'react';
import PropTypes from 'prop-types';
import Product from '../Product/Product';
import { Price } from '@boldcommerce/stacks-ui'

const LineItem = ({
  title,
  image,
  quantity,
  totalPrice,
  variants,
}) => (
  <div className="SummaryBlock CartItem" >
    <Product
      title={title}
      image={image}
      variants={variants}
    />
    <div className="CartItem__QuantityPriceWrapper">
      <div className="CartItem__ProductQuantityWrapper">
        <div className="ProductQuantity" aria-label="product quantity">
          {quantity}
        </div>
      </div>
      <div className="CartItem__ProductPrice">
        <Price amount={totalPrice} />
      </div>
    </div>
  </div>
);

LineItem.propTypes = {
  title: PropTypes.string,
  image: PropTypes.string,
  quantity: PropTypes.number,
  lineItemKey: PropTypes.string,
  variants: PropTypes.array,
};

export default LineItem;
