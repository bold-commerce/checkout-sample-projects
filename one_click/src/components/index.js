export * from './Address';
export * from './BackButton';
export * from './Billing';
export * from './BillingAddress';
export * from './Card';
export * from './CheckoutButton';
export * from './CheckoutForm';
export * from './Confirmation';
export * from './CustomerInfo';
export * from './Discount';
export * from './EmptyState';
export * from './Header';
export * from './Icons';
export * from './Index';
export * from './InventoryItem';
export * from './LineItems';
export * from './LoadingState';
export * from './Payment';
export * from '../pages/Processing';
export * from './Product';
export * from './RadioItem';
export * from './Shipping';
export * from './ShippingLines';
export * from './Summary';
