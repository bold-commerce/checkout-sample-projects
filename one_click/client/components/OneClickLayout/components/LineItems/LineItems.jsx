/* eslint-disable react/forbid-prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import LineItem from './LineItem';
import { useLineItems } from '@boldcommerce/checkout-react-components';
import { useVariants } from '../../../../hooks';

export const LineItems = ({
  readOnly,
  lineItems,
  updateLineItemQuantity,
  removeLineItem,
}) => {
  const handleVariants = useVariants();
  const lineItemList = lineItems.map((item) => (
    <LineItem
      title={item.product_data.product_title}
      description={handleVariants(item.product_data.title)}
      image={item.product_data.image_url}
      quantity={item.product_data.quantity}
      price={item.product_data.price}
      totalPrice={item.product_data.total_price}
      lineItemKey={item.product_data.line_item_key}
      onQuantityChange={(lineItemKey, value) => updateLineItemQuantity(lineItemKey, value)}
      onRemove={(lineItemKey) => removeLineItem(lineItemKey)}
      readOnly={readOnly}
      key={item.product_data.line_item_key}
    />
  ));

  return lineItemList;
};

LineItems.propTypes = {
  readOnly: PropTypes.bool,
  lineItems: PropTypes.array,
  updateLineItemQuantity: PropTypes.func,
  removeLineItem: PropTypes.func,
};

const MemoizedLineItems = React.memo(LineItems);

const LineItemsContainer = () => {
  const { data: lineItems } = useLineItems();

  return (
    <MemoizedLineItems
      lineItems={lineItems}
    />
  );
};

LineItemsContainer.propTypes = {
  readOnly: PropTypes.bool,
};

export default LineItemsContainer;
