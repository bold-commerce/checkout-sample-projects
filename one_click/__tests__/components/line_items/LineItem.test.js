import React from 'react';
import { render } from '@testing-library/react';
import { LineItem } from '../../../src/components/OneClickLayout/components/LineItems'

describe('LineItem', () => {
  test('renders LineItem component with editable quantity', () => {
    const { asFragment } = render(
      <LineItem
        title="Product A"
        image="https://via.placeholder.com/150"
        quantity={2}
        totalPrice={19998}
        lineItemKey="abcd"
        variants={["Test description right here"]}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
