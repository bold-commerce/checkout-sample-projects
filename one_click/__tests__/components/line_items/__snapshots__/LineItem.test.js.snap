// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineItem renders LineItem component with editable quantity 1`] = `
<DocumentFragment>
  <div
    class="SummaryBlock CartItem"
  >
    <div
      class="CartItem__ImageDescriptionWrapper"
    >
      <div
        class="CartItem__ProductImageWrapper"
      >
        <div
          class="stx-image"
        >
          <div
            class="stx-image--content"
          >
            <div
              class="stx-image--holder"
            >
              <img
                alt="Product A"
                class="stx-image--image"
                src="https://via.placeholder.com/150"
                title="Product A"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="CartItem__ProductDetails"
      >
        <div
          class="stx-details stx-details--left"
        >
          <div
            class="stx-details__title"
          >
            Product A
          </div>
          <div
            class="stx-details__description"
          >
            Test description right here
          </div>
        </div>
      </div>
    </div>
    <div
      class="CartItem__QuantityPriceWrapper"
    >
      <div
        class="CartItem__ProductQuantityWrapper"
      >
        <div
          aria-label="product quantity"
          class="ProductQuantity"
        >
          2
        </div>
      </div>
      <div
        class="CartItem__ProductPrice"
      >
        <div
          class="stx-price stx-price--alight-center"
        >
          <p
            aria-label="Product price"
            class="stx-price__amount"
          >
            CA$199.98
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`LineItem renders read only LineItem component 1`] = `
<DocumentFragment>
  <div
    class="SummaryBlock CartItem"
  >
    <div
      class="CartItem__ImageDescriptionWrapper"
    >
      <div
        class="CartItem__ProductImageWrapper"
      >
        <div
          class="stx-image"
        >
          <div
            class="stx-image--content"
          >
            <div
              class="stx-image--holder"
            >
              <img
                alt="Product A"
                class="stx-image--image"
                src="https://via.placeholder.com/150"
                title="Product A"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="CartItem__ProductDetails"
      >
        <div
          class="stx-details stx-details--left"
        >
          <div
            class="stx-details__title"
          >
            Product A
          </div>
          <div
            class="stx-details__description"
          >
            Test description right here
          </div>
        </div>
      </div>
    </div>
    <div
      class="CartItem__QuantityPriceWrapper"
    >
      <div
        class="CartItem__ProductQuantityWrapper"
      >
        <div
          aria-label="product quantity"
          class="ProductQuantity"
        >
          2
        </div>
      </div>
      <div
        class="CartItem__ProductPrice"
      >
        <div
          class="stx-price stx-price--alight-center"
        >
          <p
            aria-label="Product price"
            class="stx-price__amount"
          >
            CA$199.98
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
